### PR TITLE
remove use of the term "blacklist" from docs/cpp/source/Doxyfile

### LINKED ISSUE #41715: Replace blacklist/whitelist in docs/cpp/source/Doxyfile
The goal of this issue is to replace all instances of *whitelist* and *blacklist* in `docs/cpp/source/Doxyfile`. See issue #41443 for more information.

### PR DESCRIPTION
**Pull request by morrme**
As requested in #41443

### CODE REVIEW

**Comment by SplitInfinity** [comment]
Could you be more specific in the PR title? E.g.  `Remove use of the term "blacklist" from docs/cpp/source/Doxyfile`.

File: docs/cpp/source/Doxyfile [hunk 1]
```diff
@@ -74,7 +74,7 @@ CREATE_SUBDIRS         = NO
 FULL_PATH_NAMES        = YES
 # Nested folders will be ignored without this.
 RECURSIVE              = YES
-# Blacklist certain file patterns from the INPUT section.
+# Block certain file patterns from the INPUT section.
```
Comment (SplitInfinity) [inline docs/cpp/source/Doxyfile:77]: What do you think about 

```suggestion
# Exclude certain file patterns from the INPUT section.
```

as the line below suggests?

Comment (morrme) [reply docs/cpp/source/Doxyfile:77]: Sounds good!

**Review by SplitInfinity — APPROVED** [review]
Thanks for working on this!

**Pull request closed without merging**